orderer/common/server/main.go

```diff
@@ -703,15 +727,16 @@ func initializeMultichannelRegistrar(
 	bccsp bccsp.BCCSP,
 	callbacks ...channelconfig.BundleActor,
 ) *multichannel.Registrar {
-
 	registrar := multichannel.NewRegistrar(*conf, lf, signer, metricsProvider, bccsp, callbacks...)
 
 	consenters := map[string]consensus.Consenter{}
 
 	var icr etcdraft.InactiveChainRegistry
-	if conf.General.BootstrapMethod == "file" && isClusterType(bootstrapBlock, bccsp) {
-		etcdConsenter := initializeEtcdraftConsenter(consenters, conf, lf, clusterDialer, bootstrapBlock, ri, srvConf, srv, registrar, metricsProvider, bccsp)
-		icr = etcdConsenter.InactiveChainRegistry
+	if conf.General.BootstrapMethod == "file" || conf.General.BootstrapMethod == "none" {
+		if bootstrapBlock != nil && isClusterType(bootstrapBlock, bccsp) {
```

Comment:
can the bootstrapBlock be nil here? We don't have the check on the left side, so how can it be nil? Wouldn't we have crashed because we didn't have this check in the past?

Reply:
bootstrapBlock should not be nil here, however, in test _TestInitializeMultichannelRegistrar/"registrar without a system channel"_, it directly calls **initializeMultichannelRegistrar**, and passed in bootstrapBlock as nil, and BootstrapMethod as "none". In this case, the test will pass the old code as it will only check cluster type if BootstrapMethod is "file", but not the new code.

Reply:
So why not **not** pass a nil bootstrap block then? we shouldn't change the production code just to fit tests...

Reply:
It's definitely good point, we should never change production code to just fit tests. 

However, currently orderer supports an action to startup without system channel defined ([FAB-15709](https://jira.hyperledger.org/browse/FAB-15709)), which sets **BootstrapMethod** to "none" and **bootstrapBlock** to be nil. 

And in this PR, we are supporting restarting orderer without providing genesisblock, which was being requested to set **BootstrapMethod** to "none", and obtain **bootstrapBlock** from system channel ([FAB-17774](https://lists.hyperledger.org/g/fabric/message/8137)). Hence, the nil check here is to distinguish the two cases.

We can also adding a new bootstrapMethod phrase here, instead of using "none" for both use cases. I'm opening to any suggestions.